`rustc` error ANSI colour output was recently changed on nightly

`rustc` recently (between `nightly-2025-10-11` and `nightly-2025-11-11`; I didn’t bisect further) changed the specific ANSI escape sequence that it uses for bold red `error` outputs: It was `\x1b[1m\x1b[38;5;9m` (https://github.com/Canop/bacon/blob/a2efe05b1a876de0fc2b9e899ba42af84800b2e8/src/tty/mod.rs#L13) but now it is `\x1b[1m\x1b[91m`. I suspect this was done in https://github.com/rust-lang/rust/pull/147207 (see https://github.com/rust-lang/rust/pull/147207/files#diff-a7e8b42398835197f8219643e5cda042a0d9bd07f6ee6ff31b96cbe69035d6a9 for a `rustc` ui test with the same change) and that other output (e. g. warnings) was also changed.

This breaks `bacon`, especially when using `--all-targets` (as the `cargo` warning causes `bacon` to suppress all the output except for what it recognises as errors and warnings).

Reproducer:
`src/main.rs`
```rust
fn main() {
    name;
}
```

`bacon.toml`:
```toml
default_job = "check"

[jobs.check]
command = ["cargo", "check", "--all-targets", "--color", "always"]
need_stdout = false
```

`bacon` output on `nightly-2025-10-11`:
```
 t   check    1 error     1 warning

 1  error[E0425]: cannot find value `name` in this scope
  --> src/main.rs:2:5
   |
 2 |     name;
   |     ^^^^ not found in this scope

 For more information about this error, try `rustc --explain E0425`.
 error: could not compile `t` (bin "t") due to 1 previous error
 2  warning: build failed, waiting for other jobs to finish...
 error: could not compile `t` (bin "t" test) due to 1 previous error
```

Output on `nightly-2025-11-11`:
```
 t   check    1 warning

 1  warning: build failed, waiting for other jobs to finish...
 error: could not compile `t` (bin "t") due to 1 previous error
```

--- a/src/analysis/standard/standard_line_analyser.rs
+++ b/src/analysis/standard/standard_line_analyser.rs
@@ -77,12 +77,12 @@ fn analyze_line(cmd_line: &CommandOutputLine) -> LineAnalysis {
                     body.csi.as_ref(),
                     body.raw.as_ref(),
                 ) {
-                    (CSI_BOLD_RED, "error", CSI_ERROR_BODY, body_raw)
+                    (CSI_BOLD_RED | CSI_BOLD_4BIT_RED, "error", CSI_ERROR_BODY, body_raw)
                         if body_raw.starts_with(": aborting due to") =>
                     {
                         LineType::Title(Kind::Sum)
                     }
-                    (CSI_BOLD_RED, title_raw, CSI_ERROR_BODY, _)
+                    (CSI_BOLD_RED | CSI_BOLD_4BIT_RED, title_raw, CSI_ERROR_BODY, _)
                         if title_raw.starts_with("error") =>
                     {
                         LineType::Title(Kind::Error)
@@ -95,10 +95,14 @@ fn analyze_line(cmd_line: &CommandOutputLine) -> LineAnalysis {
                     (CSI_BOLD_YELLOW | CSI_BOLD_4BIT_YELLOW, "warning", _, body_raw) => {
                         determine_warning_type(body_raw, content)
                     }
-                    ("", title_raw, CSI_BOLD_BLUE, "--> ") if is_spaces(title_raw) => {
+                    ("", title_raw, CSI_BOLD_BLUE | CSI_BOLD_4BIT_BLUE, "--> ")
+                        if is_spaces(title_raw) =>
+                    {
                         LineType::Location
                     }
-                    ("", title_raw, CSI_BOLD_BLUE, "::: ") if is_spaces(title_raw) => {
+                    ("", title_raw, CSI_BOLD_BLUE | CSI_BOLD_4BIT_BLUE, "::: ")
+                        if is_spaces(title_raw) =>
+                    {
                         LineType::Location
                     }
                     ("", k, CSI_BOLD_RED | CSI_RED, "FAILED") if content.strings.len() == 2 => {

--- a/src/tty/mod.rs
+++ b/src/tty/mod.rs
@@ -11,6 +11,7 @@ pub const CSI_GREEN: &str = "\u{1b}[32m";
 
 pub const CSI_RED: &str = "\u{1b}[31m";
 pub const CSI_BOLD_RED: &str = "\u{1b}[1m\u{1b}[38;5;9m";
+pub const CSI_BOLD_4BIT_RED: &str = "\u{1b}[1m\u{1b}[91m";
 pub const CSI_BOLD_ORANGE: &str = "\u{1b}[1m\u{1b}[38;5;208m";
 pub const CSI_BOLD_GREEN: &str = "\u{1b}[1m\u{1b}[38;5;34m";
 
@@ -26,6 +27,8 @@ pub const CSI_BOLD_YELLOW: &str = "\u{1b}[1m\u{1b}[33m";
 pub const CSI_BOLD_BLUE: &str = "\u{1b}[1m\u{1b}[38;5;14m";
 #[cfg(not(windows))]
 pub const CSI_BOLD_BLUE: &str = "\u{1b}[1m\u{1b}[38;5;12m";
+#[cfg(not(windows))]
+pub const CSI_BOLD_4BIT_BLUE: &str = "\u{1b}[1m\u{1b}[94m";
 
 #[cfg(windows)]
 pub const CSI_BOLD_4BIT_YELLOW: &str = "\u{1b}[1m\u{1b}[33m";

